src/execution_result.rs

```diff
@@ -0,0 +1,60 @@
+use colored::Colorize;
+use std::{fmt, time::Duration};
+
+pub struct ExecutionResult {
+    pub runtime: Duration,
+    pub threshold: Duration,
+    pub logs: String,
+    pub output: serde_json::Value,
+}
```

Comment:
I love this approach! In the future, we plan to integrate this tool into the Shopify CLI, so having a struct like this would be very useful in allowing the caller to specify the output format (e.g. human-readable text, JSON, etc.).